### PR TITLE
AccountStats Plugin

### PR DESCRIPTION
**Pull request by rnevet**
## Description
Implemented a plugins structure, first plugin is AccountStats.
AccountStats fetches your loans history and provides a Daily Earnings Notification

## TESTING STAGE
In progress, currently seeing mismatching results when compared to exported CSV data.

## Types of changes
- [N/A] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [N/A] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] **I have read CONTRIBUTING.md**
- [X] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [X] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [X] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [N/A] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [x] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

### CODE REVIEW

**Comment by rnevet** [comment]
Related to https://github.com/BitBotFactory/poloniexlendingbot/issues/240

File: plugins/AccountStats.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        super(AccountStats, self).on_bot_init()
+        self.init_db()
+
+    def after_lending(self):
+        self.update_history()
+        self.notify_daily()
+
+    # noinspection PyAttributeOutsideInit
+    def init_db(self):
+        self.db = sqlite3.connect(r'market_data\loan_history.sqlite3')
```
Comment (Evanito) [inline plugins/AccountStats.py]: That `r` looks like a typo, can't find reference to it in the docs.

Unless it means "relative" to current directory?

Comment (rnevet) [inline plugins/AccountStats.py]: it indicates a raw string - so \ don't need to be escaped. :)

File: docs/configuration.rst [hunk 2]
```diff
@@ -295,6 +295,26 @@ Advanced logging and Web Display
...
+Plugins allow extending Bot functionality with extra features.
+To enable/disable a plugin add/remove it to the ``plugins`` list config option, example::
+
+    plugins = Plugin1, Plugin2, etc...
+
+AccountStats Plugin
+~~~~~~~~~~~~~~~~~~~
+
+The AccountStats plugin fetches all your loan history and provides statistics based on it.
+Current implementation sends a earnings summery Notification (see Notifications sections) every 24hr.
```
Comment (rnevet) [inline docs/configuration.rst]: My classic type. Thanks!

**Comment by rnevet** [comment]
@Evanito did you run the notification, was the total earnings calculation correct?

**Comment by Evanito** [comment]
It resorts to scientific notation if you only get a few satoshi in the past 24hrs, but otherwise worked like a charm.

I was unable to install `hypothesis` yet had no errors running.

**Comment by Evanito** [comment]
Shouldn't you add the `plugins =` setting to the default config?

**Comment by rnevet** [comment]
I want to stop adding optional setting to the example config, what do you think?

**Comment by Reno007** [comment]
Cool that this has now been merged! I have added the relevant line in the config file but am not getting the notification (have telegram enabled). Am I supposed to wait 24 hours? Anything else to enable?

**Comment by utdrmac** [comment]
I'm not seeing log messages indicating that the plugin has loaded.

```
May 28 08:11:45 gasgiant python[31958]: Welcome to Poloniex Lending Bot
May 28 08:11:45 gasgiant python[31958]: Started WebServer, lendingbot status available at http://10.10.10.202:8000/lendingbot.html
```

That's all I see. I can see the `market_data/loan_history.sqlite3` db was created and I can select the data inside the `history` table. But other than that, there's been no indication of any plugin activity.

**Comment by utdrmac** [comment]
The core of lendingbot just use `print` to output messages. Why is this plugin using something different (`self.log.log`)?

**Comment by rnevet** [comment]
Place the plugins line under config [BOT], the implementation is spamming you'll notice it's working

File: plugins/AccountStats.py (hunk 1 above)
Comment (chfast) [inline plugins/AccountStats.py]: You should use `/` as path separator (work on any system). Or `os.path.join()`.

**Comment by tamaskan** [comment]
Great, can't wait for the added functionality of  https://github.com/BitBotFactory/poloniexlendingbot/pull/246 and https://github.com/BitBotFactory/poloniexlendingbot/pull/114 :-)

**Pull request merged**